src/jit/flowgraph.cpp

```diff
@@ -438,6 +438,12 @@ void Compiler::fgEnsureFirstBBisScratch()
         {
             block->inheritWeight(fgFirstBB);
         }
+
+        // The first basic block will have a ref count of at least one. This corresponds to the virtual edge
+        // from the prolog to the entry block. We need to remove this virtual edge before adding the scratch
+        // block as a predecessor.
+        fgFirstBB->bbRefs--;
+        fgAddRefPred(fgFirstBB, block);
```

Comment:
Why wasn't this a problem before?

We don't always have predecessor lists built when this function is called. Seems like this would be bad in that case.